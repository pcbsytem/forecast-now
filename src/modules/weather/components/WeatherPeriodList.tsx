import React from 'react'
import { Col, FlexboxGrid, List } from 'rsuite'
import {
  WeatherPeriodListImageContainer,
  WeatherPeriodListImage,
  WeatherPeriodListDate,
  WeatherPeriodListItem
} from '../Weather.style'
import { WeatherPeriodListProps } from '../Weather.type'
import Temperature from './Temperature'
import WeatherPeriodDescription from './WeatherPeriodDescription'

const WeatherPeriodList = ({ list }: WeatherPeriodListProps) => {
  const imageAlt = 'weather description'

  return (
    <List hover style={{ width: '100%' }}>
      {list.map((listItem, index) => (
        <List.Item key={index}>
          <FlexboxGrid justify="start">
            <Temperature {...listItem} />
            <WeatherPeriodListItem
              as={Col}
              colSpan={24}
              md={20}
              screenwidth={window.screen.availWidth}
            >
              <WeatherPeriodDescription {...listItem} />
              <WeatherPeriodListDate>
                {new Date(listItem[0].startTime).toLocaleDateString()}
              </WeatherPeriodListDate>
            </WeatherPeriodListItem>
            <WeatherPeriodListImageContainer
              as={Col}
              colSpan={24}
              md={2}
              screenwidth={window.screen.availWidth}
            >
              <WeatherPeriodListImage src={listItem[0].icon} alt={imageAlt} />
            </WeatherPeriodListImageContainer>
          </FlexboxGrid>
        </List.Item>
      ))}
    </List>
  )
}

export default WeatherPeriodList
