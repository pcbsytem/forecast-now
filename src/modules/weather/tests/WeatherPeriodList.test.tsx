import React from "react";
import { render } from "@testing-library/react";
import WeatherPeriodList from "../components/WeatherPeriodList";

const mockWeatherPeriod = [
  [
    {
      number: 12,
      name: "Monday Night",
      startTime: "2022-07-04T18:00:00-04:00",
      endTime: "2022-07-05T06:00:00-04:00",
      isDaytime: false,
      temperature: 71,
      temperatureUnit: "F",
      temperatureTrend: null,
      windSpeed: "5 mph",
      windDirection: "S",
      icon: "https://api.weather.gov/icons/land/night/tsra_hi,30?size=medium",
      shortForecast: "Chance Showers And Thunderstorms",
      detailedForecast:
        "A chance of showers and thunderstorms. Partly cloudy, with a low around 71. Chance of precipitation is 30%.",
    },
    {
      number: 11,
      name: "Independence Day",
      startTime: "2022-07-04T06:00:00-04:00",
      endTime: "2022-07-04T18:00:00-04:00",
      isDaytime: true,
      temperature: 88,
      temperatureUnit: "F",
      temperatureTrend: null,
      windSpeed: "5 mph",
      windDirection: "W",
      icon: "https://api.weather.gov/icons/land/day/sct/tsra_hi,30?size=medium",
      shortForecast: "Mostly Sunny then Chance Showers And Thunderstorms",
      detailedForecast:
        "A chance of showers and thunderstorms after 2pm. Mostly sunny, with a high near 88. Chance of precipitation is 30%.",
    },
  ],
  [
    {
      number: 14,
      name: "Tuesday Night",
      startTime: "2022-07-05T18:00:00-04:00",
      endTime: "2022-07-06T06:00:00-04:00",
      isDaytime: false,
      temperature: 70,
      temperatureUnit: "F",
      temperatureTrend: null,
      windSpeed: "7 mph",
      windDirection: "SW",
      icon: "https://api.weather.gov/icons/land/night/tsra_sct,40/tsra_sct,30?size=medium",
      shortForecast: "Chance Showers And Thunderstorms",
      detailedForecast:
        "A chance of showers and thunderstorms. Mostly cloudy, with a low around 70. Chance of precipitation is 40%.",
    },
    {
      number: 13,
      name: "Tuesday",
      startTime: "2022-07-05T06:00:00-04:00",
      endTime: "2022-07-05T18:00:00-04:00",
      isDaytime: true,
      temperature: 88,
      temperatureUnit: "F",
      temperatureTrend: null,
      windSpeed: "2 to 7 mph",
      windDirection: "S",
      icon: "https://api.weather.gov/icons/land/day/rain_showers,30/tsra_hi,40?size=medium",
      shortForecast:
        "Chance Rain Showers then Chance Showers And Thunderstorms",
      detailedForecast:
        "A chance of rain showers before 8am, then a chance of showers and thunderstorms. Partly sunny, with a high near 88. Chance of precipitation is 40%.",
    },
  ],
];

describe("WeatherPeriodList component", () => {
  it("Should render correctly", () => {
    expect(
      render(<WeatherPeriodList list={mockWeatherPeriod} />),
    ).toMatchSnapshot();
  });

  it("Should render WeatherPeriodList with empty list", () => {
    expect(render(<WeatherPeriodList list={[]} />)).toMatchSnapshot();
  });
});
