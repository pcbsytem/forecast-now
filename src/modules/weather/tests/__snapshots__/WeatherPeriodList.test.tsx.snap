// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherPeriodList component Should render WeatherPeriodList with empty list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rs-list rs-list-hover"
        role="list"
        style="width: 100%;"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="rs-list rs-list-hover"
      role="list"
      style="width: 100%;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`WeatherPeriodList component Should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rs-list rs-list-hover"
        role="list"
        style="width: 100%;"
      >
        <div
          class="rs-list-item rs-list-item-md"
          role="listitem"
        >
          <div
            class="rs-flex-box-grid rs-flex-box-grid-top rs-flex-box-grid-start"
          >
            <div
              class="sc-bczRLJ iJWIhQ rs-col rs-col-md-2"
              colspan="24"
              role="gridcell"
              screenwidth="0"
            >
              <div
                class="sc-gsnTZi iiSpRq"
              >
                <b
                  class="sc-dkzDqf cybjuV"
                >
                  71
                </b>
                <b>
                  °F
                </b>
              </div>
              <div
                class="sc-gsnTZi iiSpRq"
              >
                <b
                  class="sc-dkzDqf cybjuV"
                >
                  88
                </b>
                <b>
                  °F
                </b>
              </div>
            </div>
            <div
              class="sc-iBkjds gFLrEu rs-col rs-col-md-20"
              colspan="24"
              role="gridcell"
              screenwidth="0"
            >
              <div>
                <b>
                  Independence Day
                </b>
                <div
                  class="sc-papXJ iGjbPw"
                >
                  A chance of showers and thunderstorms after 2pm. Mostly sunny, with a high near 88. Chance of precipitation is 30%.
                </div>
              </div>
              <div>
                <b>
                  Monday Night
                </b>
                <div
                  class="sc-papXJ iGjbPw"
                >
                  A chance of showers and thunderstorms. Partly cloudy, with a low around 71. Chance of precipitation is 30%.
                </div>
              </div>
              <div
                class="sc-ftvSup vRCYJ"
              >
                7/4/2022
              </div>
            </div>
            <div
              class="sc-jqUVSM hdYgGq rs-col rs-col-md-2"
              colspan="24"
              role="gridcell"
              screenwidth="0"
            >
              <img
                alt="weather description"
                class="sc-kDDrLX kDwFGe"
                src="https://api.weather.gov/icons/land/night/tsra_hi,30?size=medium"
              />
            </div>
          </div>
        </div>
        <div
          class="rs-list-item rs-list-item-md"
          role="listitem"
        >
          <div
            class="rs-flex-box-grid rs-flex-box-grid-top rs-flex-box-grid-start"
          >
            <div
              class="sc-bczRLJ iJWIhQ rs-col rs-col-md-2"
              colspan="24"
              role="gridcell"
              screenwidth="0"
            >
              <div
                class="sc-gsnTZi iiSpRq"
              >
                <b
                  class="sc-dkzDqf cybjuV"
                >
                  70
                </b>
                <b>
                  °F
                </b>
              </div>
              <div
                class="sc-gsnTZi iiSpRq"
              >
                <b
                  class="sc-dkzDqf cybjuV"
                >
                  88
                </b>
                <b>
                  °F
                </b>
              </div>
            </div>
            <div
              class="sc-iBkjds gFLrEu rs-col rs-col-md-20"
              colspan="24"
              role="gridcell"
              screenwidth="0"
            >
              <div>
                <b>
                  Tuesday
                </b>
                <div
                  class="sc-papXJ iGjbPw"
                >
                  A chance of rain showers before 8am, then a chance of showers and thunderstorms. Partly sunny, with a high near 88. Chance of precipitation is 40%.
                </div>
              </div>
              <div>
                <b>
                  Tuesday Night
                </b>
                <div
                  class="sc-papXJ iGjbPw"
                >
                  A chance of showers and thunderstorms. Mostly cloudy, with a low around 70. Chance of precipitation is 40%.
                </div>
              </div>
              <div
                class="sc-ftvSup vRCYJ"
              >
                7/5/2022
              </div>
            </div>
            <div
              class="sc-jqUVSM hdYgGq rs-col rs-col-md-2"
              colspan="24"
              role="gridcell"
              screenwidth="0"
            >
              <img
                alt="weather description"
                class="sc-kDDrLX kDwFGe"
                src="https://api.weather.gov/icons/land/night/tsra_sct,40/tsra_sct,30?size=medium"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rs-list rs-list-hover"
      role="list"
      style="width: 100%;"
    >
      <div
        class="rs-list-item rs-list-item-md"
        role="listitem"
      >
        <div
          class="rs-flex-box-grid rs-flex-box-grid-top rs-flex-box-grid-start"
        >
          <div
            class="sc-bczRLJ iJWIhQ rs-col rs-col-md-2"
            colspan="24"
            role="gridcell"
            screenwidth="0"
          >
            <div
              class="sc-gsnTZi iiSpRq"
            >
              <b
                class="sc-dkzDqf cybjuV"
              >
                71
              </b>
              <b>
                °F
              </b>
            </div>
            <div
              class="sc-gsnTZi iiSpRq"
            >
              <b
                class="sc-dkzDqf cybjuV"
              >
                88
              </b>
              <b>
                °F
              </b>
            </div>
          </div>
          <div
            class="sc-iBkjds gFLrEu rs-col rs-col-md-20"
            colspan="24"
            role="gridcell"
            screenwidth="0"
          >
            <div>
              <b>
                Independence Day
              </b>
              <div
                class="sc-papXJ iGjbPw"
              >
                A chance of showers and thunderstorms after 2pm. Mostly sunny, with a high near 88. Chance of precipitation is 30%.
              </div>
            </div>
            <div>
              <b>
                Monday Night
              </b>
              <div
                class="sc-papXJ iGjbPw"
              >
                A chance of showers and thunderstorms. Partly cloudy, with a low around 71. Chance of precipitation is 30%.
              </div>
            </div>
            <div
              class="sc-ftvSup vRCYJ"
            >
              7/4/2022
            </div>
          </div>
          <div
            class="sc-jqUVSM hdYgGq rs-col rs-col-md-2"
            colspan="24"
            role="gridcell"
            screenwidth="0"
          >
            <img
              alt="weather description"
              class="sc-kDDrLX kDwFGe"
              src="https://api.weather.gov/icons/land/night/tsra_hi,30?size=medium"
            />
          </div>
        </div>
      </div>
      <div
        class="rs-list-item rs-list-item-md"
        role="listitem"
      >
        <div
          class="rs-flex-box-grid rs-flex-box-grid-top rs-flex-box-grid-start"
        >
          <div
            class="sc-bczRLJ iJWIhQ rs-col rs-col-md-2"
            colspan="24"
            role="gridcell"
            screenwidth="0"
          >
            <div
              class="sc-gsnTZi iiSpRq"
            >
              <b
                class="sc-dkzDqf cybjuV"
              >
                70
              </b>
              <b>
                °F
              </b>
            </div>
            <div
              class="sc-gsnTZi iiSpRq"
            >
              <b
                class="sc-dkzDqf cybjuV"
              >
                88
              </b>
              <b>
                °F
              </b>
            </div>
          </div>
          <div
            class="sc-iBkjds gFLrEu rs-col rs-col-md-20"
            colspan="24"
            role="gridcell"
            screenwidth="0"
          >
            <div>
              <b>
                Tuesday
              </b>
              <div
                class="sc-papXJ iGjbPw"
              >
                A chance of rain showers before 8am, then a chance of showers and thunderstorms. Partly sunny, with a high near 88. Chance of precipitation is 40%.
              </div>
            </div>
            <div>
              <b>
                Tuesday Night
              </b>
              <div
                class="sc-papXJ iGjbPw"
              >
                A chance of showers and thunderstorms. Mostly cloudy, with a low around 70. Chance of precipitation is 40%.
              </div>
            </div>
            <div
              class="sc-ftvSup vRCYJ"
            >
              7/5/2022
            </div>
          </div>
          <div
            class="sc-jqUVSM hdYgGq rs-col rs-col-md-2"
            colspan="24"
            role="gridcell"
            screenwidth="0"
          >
            <img
              alt="weather description"
              class="sc-kDDrLX kDwFGe"
              src="https://api.weather.gov/icons/land/night/tsra_sct,40/tsra_sct,30?size=medium"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
